### ISSUE TITLE
Why RSQrq is not called RSQ if gradient is not in it?

### ISSUE THREAD

**Issue by zhzhzoo-autra** [giaf/hpipm#125]

**Comment by giaf** [comment]
You are right that the QP data `r` and `q` is normally not stored in the additional row of `RSQrq`.
However, internally for performance reasons these vectors may be copied in there before performing an operation on the whole extended matrix, see e.g. https://github.com/giaf/hpipm/blob/master/ocp_qp/x_ocp_qp_kkt.c#L71
So in a way the `rq` part only reminds of the presence of this additional work space available, as hinted here https://github.com/giaf/hpipm/blob/master/include/hpipm_d_ocp_qp.h#L58
